db/data_migrate/fetch_full_dmsids.rb

```diff
@@ -0,0 +1,77 @@
+require 'roo'
+require 'roo-xls'
+require 'csv'
+
+submission_ids = SubmissionEntry
+                 .where("LENGTH(data->>'Digital Marketplace Service ID') = 4")
+                 .where(aasm_state: 'validated')
+                 .pluck(:submission_id)
+                 .uniq
+
+entries_to_correct = {}
+
+submission_ids.each do |submission_id|
+  submission = Submission.find(submission_id)
+  file = submission.files&.first
+  next if file.nil?
```

Comment:
Were there any file.nils? how would this have happened?

Reply:
There shouldn't be, probably a paranoid guard!